core/app/[locale]/(default)/blog/page.tsx

```diff
@@ -4,9 +4,9 @@ import { notFound } from 'next/navigation';
 import { FeaturedBlogPostList } from '@/vibes/soul/sections/featured-blog-post-list';
 import { defaultPageInfo, pageInfoTransformer } from '~/data-transformers/page-info-transformer';
 
-import { getBlog, getBlogMetaData, getBlogPosts } from './page-data';
+import { BlogPostsFiltersInput, getBlog, getBlogMetaData, getBlogPosts } from './page-data';
 
-type SearchParams = Record<string, string | string[] | undefined>;
+type SearchParams = Record<string, string | string[] | undefined> & BlogPostsFiltersInput;
```

Comment:
I don't think it's safe to add types to `SearchParams` like this since a user could type anything. Instead, I think we should use `nuqs` to properly parse search params to get type safety and parsing of external inputs